client/src/modules/events/components/Actions.tsx

```diff
@@ -143,16 +146,22 @@ export const Actions = ({
   }
 
   async function onCancelAttendance() {
-    const ok = await confirm({
+    const cancelMessage: {
+      title: string;
+      body?: string;
+    } = {
       title: 'Are you sure you want to cancel your attendance?',
-    });
+    };
+    if (waitlist.length > 0) {
```

Comment:
```suggestion
    if (waitlist.length > 0 && !inviteOnly) {
```
There's one edge case - invite only event, in which case text shouldn't be displayed. For invite only event it's also not guaranteed canceling user is already off the waitlist at this point.